datastore/datastore.go

```diff
@@ -203,13 +205,26 @@ func newClient(scope string, kv string, addr string, config *store.Config, cache
 		config = &store.Config{}
 	}
 
-	addrs := strings.Split(addr, ",")
+	var addrs []string
+
+	if kv == string(store.BOLTDB) {
```

Comment:
This is why this whole path support should be done in libkv so that we don't need to add store specific logic over here. If libkv supports this, then we just need to pass the address as is and each of the libkv backend implementations will do backend specific parsing.